python/nano/src/bigdl/nano/pytorch/trainer/Trainer.py

```diff
@@ -290,37 +302,34 @@ def quantize(self, pl_model,  # remove the type requirement for type checking
                 'dynamic': 'post_training_dynamic_quant'
             }
             approach = approach_map.get(approach)
+            if accelerator is None:
+                framework = 'pytorch_{}'.format(method)
+            if accelerator == 'onnxruntime':
+                framework = "{}_{}ops".format('onnxrt', method)
 
-            quantized_model = None
-            # for framework_item in framework:
             quantizer = QuantizationINC(framework=framework, conf=conf, approach=approach,
                                         tuning_strategy=tuning_strategy,
                                         accuracy_criterion=accuracy_criterion,
                                         timeout=timeout, max_trials=max_trials)
-            if "pytorch" in framework:
-                # for 'pytorch'|'pytorch_fx'|'pytorch_ipex'
-                model: Any = pl_model  # state model to be 'Any' since we may have pl or onnx
-            else:
-                # for 'onnxrt_integerops'|'onnxrt_qlinearops'
-                pl_model = bind_onnxrt_methods(pl_model)
-                if approach == "post_training_static_quant":
-                    assert calib_dataloader, \
-                        "calib_calib_dataloader must not be None when approach is " \
-                        "post-training static quantization."
-                    pl_model.eval_onnx(input_sample=tuple(next(iter(
-                        calib_dataloader))[:-1]),
-                        file_path="model.onnx")
-                    model = pl_model.ort_infer_engine.onnx_model_fp32
-                else:
-                    assert pl_model.ort_infer_engine.onnx_model_fp32, \
-                        "Please call `eval_onnx` on model to " \
-                        "update/build your onnx structure."
-                    model = pl_model.ort_infer_engine.onnx_model_fp32
+            model = pl_model
+            if accelerator == "onnxruntime":
+                if not type(model).__name__ == 'PytorchONNXRuntimeModel':
+                    # try to establish onnx model
+                    if not input_sample:
+                        # input_sample can be a dataloader
+                        input_sample = calib_dataloader or val_dataloader
+                    model = Trainer.trace(model,
+                                          input_sample=input_sample,
+                                          accelerator='onnxruntime')
+                model = model.onnx_model
             quantized_model = quantizer.post_training_quantize(model, calib_dataloader,
```

Comment:
what is the type of the returned quantized_model?

Reply:
It's a INC defined model. For pytorch,  it should be a `PytorchModel` definded by INC in which `quantized_model.model` is a `GraphModule`. For ONNX, returned model is `ONNXModel` defined by INC in which `quantized_model.model` is a real onnx model.

Reply:
Please add some comments.